src/include/sof/probe/probe.h

```diff
@@ -8,7 +8,7 @@
 #ifndef __SOF_PROBE_PROBE_H__
 #define __SOF_PROBE_PROBE_H__
 
-#if CONFIG_PROBE
+#if CONFIG_PROBE || CONFIG_PROBE_MODULE
```

Comment:
unrelated to this PR, but why is this check even needed? This header only declares some functions, should be no harm if `CONFIG_PROBE` isn't defined either